### DISCUSSION TITLE
Receiving all applications regardless of scope

### DISCUSSION THREAD

**Discussion by MHBdev2** [mywalkb/LSPosed_mod#101] [General]
### Is your feature request related to a problem?

no

### Describe the solution you'd like

Greetings
First of all thank you very much for your amazing work, I really appreciate…
I wanted to know if there is a possibility that when connecting to an internal class of java it will be possible to receive the data of all the applications that extend that class, and not only the applications that are defined in the scope
For example
I connect to the android.app.Activity class and the onCreate function to get all the names of the Activity opened on the device, by getting param.thisObject
But in practice it only shows the applications that extend this class only if they are defined in scope even though the connection is to the parent class and yet I should receive all the applications that extend the aforementioned class.
There is a way to remove the limitation of lsposed, and get the information about all the applications that extend the aforementioned department ?

Many thanks, and sorry for my english...

### Additional context

null

**Comment by mywalkb** [comment]
If you want receive all applications, you have 2 ways, hooks all applications or hook android framework, in android framework there is ActivityController which start all activities, but I have no experience with this so I can't help you much.

**Comment by MHBdev2** [comment]
Even today I connect to internal methods that all applications use
such as android.app.Activity or android.webkit.WebView and so on
The problem is that when I want to get information about who is using these departments (such as what is the url that is now opened by using the following code
```

    Class<?> WebClass = XposedHelpers.findClass("android.webkit.WebView", lpparam.classLoader);
         findAndHookMethod(WebClass, "loadUrl", String.class, Map.class, new XC_MethodHook() 
   protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                if (param.args[0] != null) {
                    String url = (String) param.args[0];
                    XposedBridge.log("MHB_loadUrl URL hook----:  " + url);

                }
            }

```
I only receive information from the applications that are listed in the scope and not from all the applications that really expand this department
